apps/anoma_node/lib/node/transaction/backends.ex

```diff
@@ -381,4 +392,89 @@ defmodule Anoma.Node.Transaction.Backends do
 
     EventBroker.event(event)
   end
+
+  @spec cairo_resource_tx(String.t(), binary(), Noun.t()) ::
+          :ok | :error
+  defp cairo_resource_tx(node_id, id, result) do
+    with {:ok, tx} <- CTransaction.from_noun(result),
+         true <- Anoma.RM.Transaction.verify(tx),
+         true <- root_existence_check(tx, node_id, id),
+         # No need to check the commitment existence
+         true <- nullifier_existence_check(tx, node_id, id) do
+      {ct, append_roots} =
+        case Ordering.read(node_id, {id, :ct}) do
+          :absent ->
+            {CTransaction.cm_tree(),
+             MapSet.new([Anoma.Constants.default_cairo_rm_root()])}
+
+          val ->
+            {val, MapSet.new()}
+        end
+
+      {ct_new, anchor} =
+        CommitmentTree.add(ct, tx.commitments)
+
+      Ordering.add(
+        node_id,
+        {id,
+         %{
+           append: [
+             {:nullifiers, MapSet.new(tx.nullifiers)},
+             {:commitments, MapSet.new(tx.commitments)},
```

Comment:
Do you do any commitment checks? If not, why do you need to add to the commitment set here? Asking this because one way or another I don't think we should have this set around and should avoid it whenever possible.

Reply:
ah, you're right. We don't need the commitment set for shielded resources. I'll remove it.